environments/se/core/src/main/java/org/jboss/weld/environment/se/WeldContainer.java

```diff
@@ -238,6 +242,13 @@ private void fireContainerInitializedEvent() {
      */
     public Event<Object> event() {
         checkState();
+        if (event == null) {
+            synchronized (this) {
```

Comment:
Visibility - `event` is volatile and may be initialized outside the `complete()` method, the same applies to `instance` (that is the point of WELD-2371).